pyanaconda/dbus/launcher.py

```diff
@@ -0,0 +1,104 @@
+#
+# Functions to launch and test DBus session.
+#
+# Copyright (C) 2018
+# Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author(s):  Jiri Konecny <jkonecny@redhat.com>
+#
+
+import os
+
+from pyanaconda.dbus.constants import DBUS_SESSION_ADDRESS
+from pyanaconda.core.util import execWithCapture
+from pyanaconda.core.constants import ANACONDA_RUN_DIR, ANACONDA_BUS_ADDR_FILE
+
+__all__ = ["is_dbus_session_running", "start_dbus_session", "get_dbus_session_address",
+           "clean_bus_address_file"]
+
+DBUS_LAUNCH_BIN = "dbus-daemon"
+
+
+def is_dbus_session_running():
+    """Check if dbus session is running.
+
+    :returns: True if DBus is running, False otherwise
+    """
+    if os.environ.get(DBUS_SESSION_ADDRESS):
+        return True
+    elif os.path.exists(_make_bus_address_file_path()):
```

Comment:
If the file exists, but the `DBUS_STARTER_ADDRESS` is not set, anaconda will fail, no? Is the existence of the file really relevant?

Reply:
That is not `DBUS_STARTER_ADDRESS` but `DBUS_SESSION_ADDRESS` which must be set.

Reply:
My mistake. I've updated the question.

Reply:
In the case that `DBUS_SESSION_ADDRESS` is not set but the file exists means that Anaconda was restarted but it created its own session bus which is not in the OS environments -- can't be. So use this existing session bus instead of creating new one.

Reply:
So what about compromise. I will remove that `os.path.exists()` part and add comment for the whole file with warning that it is really NOT RECOMMENDED to use this in modules and it probably not work.

What do you think?